### PR TITLE
Search custom feeds

### PR DESCRIPTION
**Pull request by ansh**
Allows you to search custom feeds by their name or description

### CODE REVIEW

**Review by pfrazee — APPROVED** [review]
Looking good. I think we could squeeze slightly better perf out of it by moving the `feeds` property to `_feeds` and replacing it with a `get feeds()` accessor that runs the filtering logic against `_feeds` that's currently being applied in `search()`. Then the search function would just set a a searchQuery property that drives the `get feeds()` accessor, and you can drop some of the refreshes in the view when the search query state changes.

**Comment by pfrazee** [comment]
Oh I accidentally marked that as approved instead of just a comment. Sec while I test.

**Comment by ansh** [comment]
Good idea. Let me do that…

**Comment by ansh** [comment]
Waiting on confirmation from @estrattonbailey about implementing this on the backend... Steady lads 🖖🏽

**Comment by ansh** [comment]
@pfrazee Need a re-review since now search is taking place on the backend

**Comment by pfrazee** [comment]
Scratch that -- the validation was failing clientside because I hadnt run yarn

File: src/view/screens/DiscoverFeeds.tsx [hunk 1]
```diff
@@ -22,6 +23,33 @@ export const DiscoverFeedsScreen = withAuthRequired(
...
     const feeds = React.useMemo(() => new FeedsDiscoveryModel(store), [store])
 
+    // search stuff
+    const [isInputFocused, setIsInputFocused] = React.useState<boolean>(false)
+    const [query, setQuery] = React.useState<string>('')
+    const onChangeQuery = React.useCallback(
+      (text: string) => {
+        setQuery(text)
+        if (text.length > 1) {
+          feeds.search(text)
```
Comment (pfrazee) [inline src/view/screens/DiscoverFeeds.tsx:33]: We need to debounce this query -- it's currently running on every change and the results kind of oddly update as you go, often with some lag.

**Review by pfrazee — CHANGES_REQUESTED** [review]

File: src/view/screens/DiscoverFeeds.tsx (hunk 1 above)
Comment (ansh) [reply src/view/screens/DiscoverFeeds.tsx]: Fixed.

**Pull request merged**